js/app/test/blocks_chained_events.spec.ts

```diff
@@ -0,0 +1,69 @@
+import { test, expect } from "@gradio/tootils";
+
+test(".success event runs after function successfully completes. .success should not run if function fails", async ({
+	page
+}) => {
+	const textbox = page.getByLabel("Result");
+	await expect(textbox).toHaveValue("");
+
+	await Promise.all([
+		page.click("text=Trigger Failure"),
+		page.waitForResponse("**/run/predict")
+	]);
+	expect(textbox).toHaveValue("");
+
+	await Promise.all([
+		page.click("text=Trigger Success"),
+		page.waitForResponse("**/run/predict")
+	]);
+
+	expect(textbox).toHaveValue("Success event triggered");
+});
+
+test("Consecutive .success event is triggered successfully", async ({
+	page
+}) => {
+	const textbox = page.getByLabel("Consecutive Event");
+	const first = page.getByLabel("Result");
+
+	await page.click("text=Trigger Consecutive Success");
+	let count = 0;
+	await page.waitForResponse((url) => {
+		if (url.url().endsWith("run/predict")) {
+			count += 1;
+			return count == 3;
+		}
+		return false;
```

Comment:
Sorry I don't quite follow what this count is doing, could you explain?

Reply:
It looks like it's also tripping up the CI

Reply:
We are making sure that the predict endpoint is called a specific number of time before continuing. I'm trying to figure out the flake, it passes locally.

Reply:
Ah got it

Reply:
I think this could be a race condition since the events could be processed at varying times, although they are likely to happen one after the other there could be a chance that they dont. Hard to say without knowing more about how playwright schedules work. I am trying an alternative.

Reply:
I was wrong but changed it anyway, new way is cleaner.